src/evm-block-extractor/src/database/postgres_db_client.rs

```diff
@@ -42,13 +42,16 @@ impl PostgresDbClient {
     }
 
     async fn insert_key_value_data<D: Serialize>(&self, key: &str, data: D) -> anyhow::Result<()> {
-        sqlx::query("INSERT INTO EVM_KEY_VALUE_DATA (key, data) VALUES ($1, $2)")
-            .bind(key)
-            .bind(serde_json::to_value(data)?)
-            .execute(&self.pool)
-            .await
-            .map_err(|e| anyhow::anyhow!("Error inserting value data for key {}: {:?}", key, e))
-            .map(|_| ())
+        sqlx::query(
+            "INSERT INTO EVM_KEY_VALUE_DATA (key, data) VALUES ($1, $2)
+            ON CONFLICT (key) DO UPDATE SET data = $2",
```

Comment:
An database insert should not silently override data unless explicitly stated. Better to create a new fn or to make it explicit through an argument

Reply:
Fixed